modules/dhcp/providers/server/isc.rb

```diff
@@ -78,13 +78,10 @@ def parse_config_and_leases_for_records
             opts.merge!(parse_record_options(data))
           end
         end
-        begin
-          subnet = @service.find_subnet(opts[:ip])
-          next unless subnet
-          ret_val << Proxy::DHCP::Reservation.new(opts.merge(:subnet => subnet))
-        rescue Exception => e
-          logger.warn "skipped #{hostname} - #{e}"
-        end
+        next if !opts.has_key?(:ip)
```

Comment:
Yeah, we shouldn't skip deleted entries, otherwise memory store may end up being initialized with stale records. However, It's not possible to create a reservation without a valid subnet at the moment: there a validator in the base class that will throw an exception if the subnet is missing (https://github.com/theforeman/smart-proxy/blob/develop/modules/dhcp/record.rb#L14). I think splitting Reservation class into Reservation and DeletedReservation (where subnet is optional) would provide a clean-ish solution.